imap_processing/cdf/tests/test_cdf_attribute_manager.py

```diff
@@ -1,58 +1,351 @@
+        == "IMAP>Interstellar Mapping and Acceleration Probe"
+    )
     assert (
-        cdf_manager.global_attributes["Mission_group"]
+        cdf_manager._global_attributes["Discipline"]
+        == "Solar Physics>Heliospheric Physics"
+    )
+    assert (
+        cdf_manager._global_attributes["Mission_group"]
         == "IMAP>Interstellar Mapping and Acceleration Probe"
     )
+    assert cdf_manager._global_attributes["PI_name"] == "Dr. David J. McComas"
+    assert (
+        cdf_manager._global_attributes["PI_affiliation"]
+        == "Princeton Plasma Physics Laboratory, 100 Stellarator Road, "
+        "Princeton, NJ 08540"
+    )
+    assert (
+        cdf_manager._global_attributes["File_naming_convention"]
+        == "source_descriptor_datatype_yyyyMMdd_vNNN"
+    )
+    # The following test should fail because "DOI" is not an attribute in
+    #   imap_default_global_cdf_attrs.yaml
+    with pytest.raises(KeyError):
+        assert cdf_manager._global_attributes["DOI"] == "test"
 
-    # Load additional global attributes
-    cdf_manager.load_global_attributes("imap_mag_global_cdf_attrs.yaml")
+    # Load in different data
+    cdf_manager.source_dir = Path(__file__).parent.parent / "tests"
+    cdf_manager.load_global_attributes("imap_default_global_test_cdf_attrs.yaml")
+
+    # Testing attributes carried over
+    assert (
+        cdf_manager._global_attributes["File_naming_convention"]
+        == "source_descriptor_datatype_yyyyMMdd_vNNN"
+    )
+    assert (
+        cdf_manager._global_attributes["Discipline"]
+        == "Solar Physics>Heliospheric Physics"
+    )
 
+    # Testing attributes newly loaded
+    assert cdf_manager._global_attributes["Project"] == "STP>Solar-Terrestrial Physics"
     assert (
-        cdf_manager.global_attributes["Mission_group"]
+        cdf_manager._global_attributes["Source_name"]
         == "IMAP>Interstellar Mapping and Acceleration Probe"
     )
+    assert cdf_manager._global_attributes["Mission_group"] == "Dysfunctional Cats"
+    assert cdf_manager._global_attributes["PI_name"] == "Ana Manica"
+    assert cdf_manager._global_attributes["PI_affiliation"] == "LASP, CU"
+    assert cdf_manager._global_attributes["Data_version"] == 2
+    assert cdf_manager._global_attributes["DOI"] == "test"
 
-    assert cdf_manager.global_attributes["Descriptor"] == "MAG>Magnetometer"
+    # Testing that everything loaded into the global attrs is present in
+    #   the global attrs schema
+    for attr in cdf_manager._global_attributes.keys():
+        assert attr in cdf_manager.global_attribute_schema.keys()
 
-    mag_l1a_global_attrs = cdf_manager.get_global_attributes("imap_mag_l1a_norm-raw")
 
+def test_get_global_attributes():
+    """
+    Test function that covers:
+        get_global_attributes
+    """
+
+    # Initialize CdfAttributeManager object which loads in default info
+    cdf_manager = CdfAttributeManager(Path(__file__).parent.parent / "config")
```

Comment:
Is there a reason to first load the default (not for testing) schema and default global attributes? It makes this test a bit confusing to have that cross pollination that isn't very obvious. Can the test schema be used instead so that this is totally isolated to using test specific schema and global attribute defaults?

This comment applies to all of the test functions in this file.

Reply:
I believe the reason we are using the default schema for these tests is because they are actually loaded into the CdfAttributeManager class from the `__init__` function. I would have to change the code here if I wanted to use test schema, which I have been told is not ideal.

Reply:
I prefer to use the real schema against the test data so we can avoid the real schema and test schema from getting out of sync, so I asked Ana to do it this way. 

But, since you got a comment on it, can you add a short comment in the code explaining why you're loading in the default schema?

Reply:
@maxinelasp, Does this mean that you are considering these tests a validation of the real schema? I think of unit tests as only covering the code implementation, not validating the schema.

Reply:
Yeah, good question. I can see either argument, but in this case, I think it makes sense to have the real schema be part of the unit tests because it's closely tied to the code. But, I can definitely see the philosophical argument that it's not really part of the "unit" so it shouldn't be tested.